pkg/apis/kappctrl/v1alpha1/types_fetch.go

```diff
@@ -103,6 +112,9 @@ type AppFetchGit struct {
 	// Skip lfs download (optional)
 	// +optional
 	LFSSkipSmudge bool `json:"lfsSkipSmudge,omitempty" protobuf:"varint,5,opt,name=lfsSkipSmudge"`
+	// Specify the destination directory name for each fetch source
+	// +optional
+	DestDirName string `json:"destDirName,omitempty" protobuf:"bytes,7,opt,name=destDirName"`
```

Comment:
we should probably have a brief RFC for names for this property that includes some of our more active community members - the biggest constraint was basically that for _some_ fetch types there's already a `subPath` which refers to a path in the source (git repo or http server), so it seemed best to at least attempt to indicate that this is a path that'll exist during templating, not a path that's part of the source.